text/0000-slice-copy-fill.md

```diff
@@ -0,0 +1,83 @@
+- Feature Name: slice\_copy\_fill
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Safe `memcpy` from one slice to another of the same type and length, and a safe
+`memset` of a slice of type `T: Copy`.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the only way to quickly copy from one non-`u8` slice to another is to
+use a loop, or unsafe methods like `std::ptr::copy_nonoverlapping`. This allows
+us to guarantee a `memcpy` for `Copy` types, and is safe. The only way to
+`memset` a slice, currently, is a loop, and we should expose a method to allow
+people to do this. This also completely gets rid of the point of
+`std::slice::bytes`, which means we can remove this deprecated and useless
+module.
```

Comment:
This motivation may want to be update now that `std::slice::bytes` has been entirely deprecated (this was a recent change though)